app/notification_test.go

```diff
@@ -817,3 +819,287 @@ func TestDoesStatusAllowPushNotification(t *testing.T) {
 		t.Fatal("Should have been false")
 	}
 }
+
+func TestGetDirectMessageNotificationEmailSubject(t *testing.T) {
+	Setup()
+	expectedPrefix := "[http://localhost:8065] New Direct Message from sender on"
+	post := &model.Post{
+		CreateAt: 1501804801000,
+	}
+	translateFunc := utils.GetUserTranslations("en")
+	subject := getDirectMessageNotificationEmailSubject(post, translateFunc, "http://localhost:8065", "sender")
+	if !strings.HasPrefix(subject, expectedPrefix) {
+		t.Fatal("Expected subject line prefix '" + expectedPrefix + "', got " + subject)
+	}
+}
+
+func TestGetNotificationEmailSubject(t *testing.T) {
+	Setup()
+	expectedPrefix := "[http://localhost:8065] Notification in team on"
+	post := &model.Post{
+		CreateAt: 1501804801000,
+	}
+	translateFunc := utils.GetUserTranslations("en")
+	subject := getNotificationEmailSubject(post, translateFunc, "http://localhost:8065", "team")
+	if !strings.HasPrefix(subject, expectedPrefix) {
+		t.Fatal("Expected subject line prefix '" + expectedPrefix + "', got " + subject)
+	}
+}
+
```

Comment:
The remaining tests in this file aren't super smart - they're just checking to see if certain strings are included in the email body. I didn't want to get into complex regex or anything like that in case we decide to tweak the templates. Instead, I'm just making sure that the important information is or is not included.